### PR TITLE
feat(Modal): remove deprecated mode prop

### PR DESCRIPTION
**Pull request by langz**
The following PR removes the deprecated `mode` property from the Modal component.
In regards to this change, this PR:
- Updates the documentation.
- Replaces all internal occurrences/usage of Modal:
- - `<Modal mode="custom"/>` to be `<Modal/>`
- - `<Modal mode="dialog"/>` to be `<Dialog/>`
- - `<Modal>` to be `<Dialog/> `
- - `<Modal mode="drawer"/>` to be `<Drawer/>`
- "Updates/Fixes" the existing Modal tests, so that they still work as before/previously.

This PR does not, at the moment, fix up in the existing "dependencies" between Modal and Drawer, and Modal and Dialog. 
To me it's still blurred lines regarding which components uses which props, and which components is responsible for what, and that this is a bit intertwined.  What do I mean by that?
- As the existing Modal(pre this change) was mostly used with `mode` or as a dialog(without `mode`), a lot(most) of the existing tests for Modal still only make sense in the context of a dialog or modal. To rewrite the tests to cater for the "new modal"(written when dialog and drawer was separated from modal and the custom mode came along), I think one would have to really understand what the "new modal"(custom) actually is and what it offers. So there exists some stories and tests in "modal files" now, which I think could be move to other places(dialog and/or drawer).
- Both Dialog and Drawer inherits all Modals props. Is this correct? Like does it make sense for a Drawer to have the following prop `dialogTitle / dialog_title`, which it inherits from Modal? I don't know, but it sounds and seems specific to the dialog...
As the previous approach when using Modal's mode to separate between dialog and drawer, it made sense that it would exist unique properties that only made sense for dialog and not when a drawer, etc. But now when Dialog and Drawer was separated from Modal, I think it would make more sense for Dialog and Drawer to not inherit all props from Modal, but rather only make the needed props of the Modal for the given component(Dialog or Drawer) available through their own component(Dialog or Drawer) props. To more easily actually understand which component needs/use each prop.
- I'm also not sure if the existing docs https://eufemia.dnb.no/uilib/components/modal/info/#accessibility made/makes sense for the "new modal": _4. Disable the body scroll possibility._. As far as I can see in the "new modal"(mode custom), https://eufemia.dnb.no/uilib/components/modal/demos/#mode-custom, there's a scroll bar on the right(body scroll possibility)... I believe this is because the "new modal"(mode custom) doesn't use the ScrollView, but should it(docs says so, code says no)? I don't know.

So to sum this up, the following PR(as of now) removes the deprecated `mode` property for Modal, but there's still great improvements to be made in the codebase regarding the Modal and its relationship to Dialog and Drawer, as the code shows signs of left over code and logic from before they were split into two separate components. 

I think we can and should continue to improve upon this, but that could also be done in a new PR as I feel that this is sort of a nice clear increment that only removes the `mode` property and makes the necessary changes related to this.

BREAKING CHANGE:

- Modal: Remove deprecated `mode` prop – it will default to `custom` mode. Use rather the `Dialog` or the `Drawer` component.

### CODE REVIEW

File: packages/dnb-eufemia/src/components/modal/__tests__/Modal.test.tsx [hunk 1]
```diff
@@ -271,24 +273,9 @@ describe('Modal component', () => {
     })
     expect(testTriggeredBy).toBe(null)
 
-    Comp.find('div.dnb-dialog').simulate('keyDown', {
```
Comment (langz) [inline packages/dnb-eufemia/src/components/modal/__tests__/Modal.test.tsx:274]: This test is not relevant anymore for the "new modal"(previously custom mode), as the onKeyDownHandler which this is testing, is only "used" in DialogContent and DrawerContent(hence I've moved this test to the tests for DialogContent and DrawerContent)

**Review by tujoworker — APPROVED** [review]
💯 Nice! I think that's a good decision. And as you said in the description, we may iterate on it.

File: packages/dnb-eufemia/src/components/modal/ModalContent.tsx [hunk 2]
```diff
@@ -352,7 +349,6 @@ export default class ModalContent extends React.PureComponent<
 
   render() {
     const {
-      mode = 'modal',
```
[the remark this answers is not in the document]
Comment (langz) [reply packages/dnb-eufemia/src/components/modal/ModalContent.tsx:355]: Yes, it's a breaking change 👍 
Previously the "default Modal" would return a `Dialog`, now after this change the "default Modal" will return a "Modal" previously know as a "custom Modal"(modal with mode custom)

File: packages/dnb-design-system-portal/src/docs/uilib/components/modal/visual-tests/Examples.js [hunk 3]
```diff
@@ -10,14 +10,13 @@ export const ModalExampleDrawerHeader = () => (
   <ComponentBox data-visual-test="modal-drawer-header">
     {
       /* jsx */ `
-<Modal
-  mode="drawer"
```
[the remark this answers is not in the document]
Comment (langz) [reply packages/dnb-design-system-portal/src/docs/uilib/components/modal/visual-tests/Examples.js:14]: Yupp yupp, I'll do so 🙏

File: packages/dnb-eufemia/src/components/modal/ModalContent.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (tujoworker) [reply packages/dnb-eufemia/src/components/modal/ModalContent.tsx:355]: I'v added the needed message in the PR description. It will be used when we use squash merge.

Screenshot 2023-02-20 at 11 10 54
